Fix optimize cache headers for static assets

Fireentity opened the pull request:
**Description**
I've disabled Netlify's default caching (mergeCachingHeaders: false) to implement a more granular policy. This change sets a long-term (1-year, immutable) cache for all hashed static assets (JS, CSS, images, fonts). Conversely, HTML and Gatsby's page-data files are now set to must-revalidate to ensure users always receive fresh content.

This approach aligns with caching best practices. As part of this, I also moved the global security headers (CSP, X-Frame-Options) to the correct allPageHeaders setting.

References:

https://developer.chrome.com/docs/lighthouse/performance/uses-long-cache-ttl
https://jakearchibald.com/2016/caching-best-practices/
https://simonhearne.com/2022/caching-header-best-practices/

This PR fixes #6930

**Notes for Reviewers**
I'm addressing other open issues in separate PRs. Since this one affects caching for all pages, I felt it was best to keep it focused.

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 

<!--
Thank you for contributing to Layer5 projects! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR. 
3. Sign your commits

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

Fireentity committed:
fix: optimize cache headers for static assets

Disables Netlify's default caching headers (mergeCachingHeaders: false) to implement a more granular and performant caching strategy. This change sets long-term immutable caching (1 year) for static assets like JS, CSS, fonts, and images.

At the same time, it ensues HTML and page-data files are not cached (must-revalidate) to prevent stale content. Finally, the security headers (X-Frame-Options, CSP) are moved to the 'allPageHeaders' option for proper global application.

Signed-off-by: Lorenzo Croce <lorenzo.croce@arenastudios.it>

leecalcote reviewed (commented):
Thanks for evaluating the site's performance in this way, @Fireentity. 👍

leecalcote commented on gatsby-config.js:
```diff
@@ -21,13 +21,56 @@ module.exports = {
+          ],
+          "/*.ttf": [
+            "Cache-Control: public, max-age=31536000, immutable",
+          ],
```
Good. This is helpful in that our font files don't change.

leecalcote reviewed (commented):
More thought needs to be given to the extended caching duration proposed for some types of content.

The font file section is good to go. The rest is an open question that needs to be justified for each type of content.

Rajesh-Nagarajan-11 commented:
Thank you for your contribution!
Let's discuss this during the website call  tomorrow  ( 17/11/2025 ) at 6:30 PM IST | 8 AM CT

Add it as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit?usp=drivesdk), if you would :slightly_smiling_face:

Rajesh-Nagarajan-11 committed:
Merge branch 'master' into fix/6930-blog-page-performance

Fireentity committed:
chore: reduce cache max-age for JS and CSS files

Update gatsby-config.js to lower the Cache-Control max-age from 1 year to 30 days (2592000 seconds) for .js and .css assets.

Signed-off-by: Lorenzo Croce <lorenzo.croce@arenastudios.it>

Fireentity commented on gatsby-config.js:
```diff
@@ -21,13 +21,56 @@ module.exports = {
+          "/*.js": [
+            "Cache-Control: public, max-age=31536000, immutable",
+          ],
+          "/*.css": [
```
Specifically, I have configured 
HTML and page-data to always revalidate (max-age=0) so users immediately see new deployments
CSS and JavaScript files are now set to 30 days (max-age=2592000) to prevent cache bloat. 
Fonts and images remain at 1 year (max-age=31536000) since these assets rarely change

Fireentity committed:
fix(cache): disable long-term caching for static assets

Update Cache-Control headers for the /static/ folder to must-revalidate. Since files in this directory do not use content hashing, long-term caching prevented users from seeing immediate updates.

Signed-off-by: Lorenzo Croce <lorenzo.croce@arenastudios.it>

l5io commented:
🚀 Preview for commit 4a1b5efb022bedda7531d271a1ac9fdc1fbfe18d at: https://69238de481038ad905b0727a--layer5.netlify.app

leecalcote committed:
Merge branch 'master' into fix/6930-blog-page-performance

l5io commented:
🚀 Preview for commit ad28e0b80b00adfa7f5d1fe089e1f751b52268dc at: https://692d443c290bfdacd26849fb--layer5.netlify.app

Rajesh-Nagarajan-11 committed:
Merge branch 'master' into fix/6930-blog-page-performance

l5io commented:
🚀 Preview for commit bc65c220dee8c8ff90eb30080264229ecef3be0e at: https://692d521cd8a8d81349e20366--layer5.netlify.app

Rajesh-Nagarajan-11 committed:
Merge branch 'master' into fix/6930-blog-page-performance
